ingest/mongo_connection.py

```diff
@@ -76,40 +76,51 @@ def wrapper(*args, **kwargs):
                 else:
                     raise e
             except BulkWriteError as bwe:
-
                 if attempt < MAX_ATTEMPTS - 1:
                     dev_logger.warning(
                         "Batch ops error occurred. Reinsert attempt %s.", str(attempt)
                     )
-                    error_docs = bwe.details["writeErrors"]
+                    error = bwe.details["writeErrors"]
                     # Check error code to see if any failures are due to violating a unique index (error code 11000)
                     # and discard those documents before retrying
-                    filtered_docs = discard_inserted_documents(error_docs, args[0])
+                    filtered_docs = discard_inserted_documents(error, args[0])
                     if len(filtered_docs) > 0:
                         args[0] = filtered_docs
                         retry(attempt)
                     else:
                         return args[0]
                 else:
-                    dev_logger.debug(str(bwe.details))
+                    log_error_without_values(bwe)
                     raise bwe
 
     return wrapper
 
+def log_error_without_values(error):
+    """Remove 'values' array from log messages as this is usually very long and makes logs difficult to read
+
+    :param error: BulkWriteError from failed transaction
+    """
+    for entry in error.details["writeErrors"]:
+        if 'values' in entry['op']:
+            entry['op']['values'] = '<filtered>'
+    dev_logger.debug(str(error.details))
 
-def discard_inserted_documents(error_documents, original_documents):
+def discard_inserted_documents(errors, original_documents):
     """Discard any documents that have already been inserted which are violating index constraints
        such documents will have an error code of 11000 for a DuplicateKey error
        from https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml#L467
+       Uses Mongo query to get existing names and filters against documents to be inserted as
+       only first error is returned, not all possible errors
 
        Parameters:
-           error_documents (List[Dict]): list of documents that failed to insert in original transaction
-           original_documents (List[Dict]): list of documents from original transaction that failed
-           error_code (Int): error status code to filter on
+           errors (List<Dict>): Documents that failed to insert from transaction
```

Comment:
Closer to Python convention for documenting types, i.e. closer to [type hints](https://docs.python.org/3/library/typing.html):

```suggestion
           errors (List[Dict]): Documents that failed to insert from transaction
```

Reply:
[14532dc](https://github.com/broadinstitute/scp-ingest-pipeline/pull/271/commits/14532dcd67b2b4aecc1e19c923883cef6352c913)